pkg/oc/cli/cmd/create/deploymentconfig.go

```diff
@@ -122,6 +125,21 @@ func (o *CreateDeploymentConfigOptions) Complete(cmd *cobra.Command, f *clientcm
 	return nil
 }
 
+// parseImageName parses the image strings and returns just the image name
+func parseImageName(imageString string) string {
+	// Retain just the image name
+	imageSplit := strings.Split(imageString, "/")
+	name := imageSplit[len(imageSplit)-1]
+	// Remove any tag or hash
+	if strings.Contains(name, ":") {
```

Comment:
This code is based on `kubectl create deployment`. However, upstream seems to have an issue https://github.com/kubernetes/kubernetes/issues/62252 so here has a minor update which removes `else if`.

Reply:
Please don't parse the image name like this.  We should be using the real name parsing logic that we have elsewhere.

/hold